compiler/test/dotty/tools/backend/jvm/DottyBytecodeTests.scala

```diff
@@ -565,4 +565,19 @@ class TestBCode extends DottyBytecodeTest {
       assertEquals(14, instructionsFromMethod(method).size)
     }
   }
+
+  @Test def objectsAreFinal = {
+    val source =
+      """class Test {
+        |  object Foo
```

Comment:
I would test a few more cases (top-level object, object in object, object in local block).